File: asignaciones_y_variables_05.c
/*
* 
* Título: EJERCICIOS DE VARIABLES Y ASIGNACIONES 5
* Consigna: Ingresar dos lados de un triángulo rectángulo y calcular, la hipotenusa, el perímetro, la superficie.
*	Alumno: Nicolás Esteban Olmedo
*
*/

/* Directivas del preprocesador */
#include <stdio.h>
#include <math.h>

/* Macros del procesador */
/* Definiciones globales */
	/* Declaración de variables globales */
	/* Declaración de funciones */
void clrscr(void);
float calculo_hip(float arg0, float arg1);
float calculo_perim(float arg0, float arg1, float arg2);
float calculo_area(float arg0, float arg1);

/* Función principal */
int main() {
	clrscr();
	/* Declaración de variables locales */
	float cateto0, cateto1, hipotenusa, perimetro, area;
	
	/* Input de data */
	printf("Introducir cateto los dos catetos: ");
	scanf("%f" "%f", &cateto0, &cateto1);
	
	/* Procesamiento de data */
	hipotenusa = calculo_hip(cateto0, cateto1);
	perimetro = calculo_perim(cateto0, cateto1, hipotenusa);
	area = calculo_area(cateto0, cateto1);
	
	/* Output de data */
	printf("Hipotenusa = %f\nPerímetro = %f\nArea = %f\n", hipotenusa, perimetro, area);
	return 0;
	
}
/* Otras funciones */
	/* Definición clrscr() */
void clrscr(void) {
	system("clear");
}

	/* Definición calculo_hip() */
float calculo_hip(float arg0, float arg1) {
	float res;
	res = sqrt(pow(arg0,2)+pow(arg1,2));
	return res;
}

float calculo_perim(float arg0, float arg1, float arg2) {
	float res;
	res = arg0 + arg1 + arg2;
	return res;
}

float calculo_area(float arg0, float arg1) {
	float res;
	res = (arg0*arg1)/2;
	return res;
}

File: C#files/C#practice/src/asignaciones_y_variables_05.cs
/*
 * Ingresar dos lados de un triángulo rectángulo y calcular, la hipotenusa, el perímetro, la superficie.
 */

using System;

namespace AsignacionesVariables05 {
	class Program {
		static void Main(string[] args) {
			float ladoA, ladoB, hipotenusa, perimetro, superficie;
			string tempVal;

			//Input de data
			Console.Write("Ingresar lado A: ");
			tempVal = Console.ReadLine();
			ladoA = Convert.ToSingle(tempVal);


			Console.Write("Ingresar lado B: ");
			tempVal = Console.ReadLine();
			ladoB = Convert.ToSingle(tempVal);

			//Procesamiento de data
			hipotenusa = CalculoHipotenusa(ladoA, ladoB);
			perimetro = CalculoPerimetro(ladoA, ladoB, hipotenusa);
			superficie = CalculoSuperficie(ladoA, ladoB);

			//Output de data
			Console.WriteLine("Hipotenusa = " + hipotenusa);
			Console.WriteLine("Perímetro = " + perimetro);
			Console.WriteLine("Superficie = " + superficie);
		}

		private static float CalculoHipotenusa(float a, float b) {
			//Need help with this one
			float tempRes = (float)Math.Sqrt((float)Math.Pow(a,2) + (float)Math.Pow(b,2));
			return tempRes;
		}

		private static float CalculoPerimetro(float a, float b, float c) {
			float tempRes = a + b + c;
			return tempRes;
		}

		private static float CalculoSuperficie(float a, float b) {
			float tempRes = (a+b)/2;
			return tempRes;
		}
	}
}
